{
  Throwable thrown=null;
  try {
    a.close();
  }
 catch (  Throwable e) {
    thrown=e;
  }
  try {
    b.close();
  }
 catch (  Throwable e) {
    if (thrown == null)     thrown=e;
  }
  if (thrown == null)   return;
  if (thrown instanceof IOException)   throw (IOException)thrown;
  if (thrown instanceof RuntimeException)   throw (RuntimeException)thrown;
  if (thrown instanceof Error)   throw (Error)thrown;
  throw new AssertionError(thrown);
}
